src/services/products.js

```diff
@@ -0,0 +1,110 @@
+const { products } = require('../db');
+
+async function createProduct(req) {
+  const { type, color, price = 0, quantity = 1 } = req.body;
+
+  if (!type) {
+    const err = new Error('ERROR: No product type defined');
+    err.status = 400;
+    throw err;
+  }
```

Comment:
I can get here without `Content-Type: application/json` header and as a result I get `ERROR: No product type defined` even though the body was correct.